lib/Driver/Compilation.cpp

```diff
@@ -483,6 +515,27 @@ namespace driver {
       return res;
     }
 
+    void
+    emitParseableOutputForFinishedJob(ProcessId Pid, int ReturnCode,
+                                      StringRef Output,
+                                      const Job *FinishedCmd,
+                                      TaskProcessInformation ProcInfo) {
+      FinishedCmd->forEachContainedJobAndPID(Pid, [&](const Job *J,
```

Comment:
This looks good, but there is a small potential for confusion remaining: the function is called `emitParseableOutputForFinishedJob` yet it includes an iterator: `forEachContainedJobAndPID`. The latter suggests that it is emitting parseable output for many jobs, but the former implies emitting output for *one* job. Is the problem an ambiguity in "job"? Are there better names? Or perhaps just a comment?? Or am I splitting hairs?

Reply:
It does say "for each *contained* job", so I don't think it's *wrong,* per se. The correct parseable output for a batch job includes the parseable output for all the sub-jobs. But I see what you mean.

Reply:
I don't think it's hugely ambiguous but I get the point; renamed to `emitParseableOutputForEachFinishedJob`, which at least puts some multiplicity / each-ness in the name.